src/core/linux/SDL_dbus.c

```diff
@@ -534,6 +534,62 @@ char *SDL_DBus_GetLocalMachineId(void)
 
     return NULL;
 }
+
+/*
+ * Convert file drops with mime type "application/vnd.portal.filetransfer" to file paths
+ * Result must be freed with dbus->free_string_array().
+ * https://flatpak.github.io/xdg-desktop-portal/#gdbus-method-org-freedesktop-portal-FileTransfer.RetrieveFiles
+ */
+char **SDL_DBus_GetPortalFilePath(char *key, int *file_path_count)
+{
+    DBusError err;
+    char** file_paths = NULL;
+    DBusMessageIter iter, iterDict;
+
+    DBusMessage *reply = NULL;
+    DBusMessage *msg = dbus.message_new_method_call("org.freedesktop.portal.Documents",    /* Node */
+                                                    "/org/freedesktop/portal/documents",   /* Path */
+                                                    "org.freedesktop.portal.FileTransfer", /* Interface */
+                                                    "RetrieveFiles");                      /* Method */
+
+    if (!dbus.session_conn) {
+        /* We either lost connection to the session bus or were not able to
+         * load the D-Bus library at all. */
+        return NULL;
+    }
+
+    dbus.error_init(&err);
+
+    dbus.message_append_args(msg, DBUS_TYPE_STRING, &key, DBUS_TYPE_INVALID);
+
+    dbus.message_iter_init_append(msg, &iter);
+    if (!dbus.message_iter_open_container(&iter, DBUS_TYPE_ARRAY, "{sv}", &iterDict) || !dbus.message_iter_close_container(&iter,  &iterDict)) {
+        dbus.message_unref(msg);
```

Comment:
I think we should have a SDL_SetError("Out of memory") here

Reply:
I think the probability of this error is so low it's not worth it handling further. Allocating an empty dictionary is unlikely to run out of memory.

Reply:
In general we try to handle every error, no matter how unlikely. If it's possible, it'll happen on a customer system. We have a handy SDL_OutOfMemory() function to standardize the error message for this.

Reply:
Okay, I've added `SDL_OutOfMemory`.